packages/react-async/src/useAsync.spec.js

```diff
@@ -250,4 +250,54 @@ describe("useFetch", () => {
       expect.objectContaining({ preventDefault: expect.any(Function) })
     )
   })
+
+  test("defer=true allows using the promise", () => {
+    let check = 0
+    const component = (
+      <Fetch input="/test" options={{ defer: true }}>
+        {({ run, promise }) => (
+          <button
+            onClick={() => {
+              run()
+              promise
```

Comment:
Please consider that `promise` is only created __after__ invoking `run()`. So in this case it's correct that `promise` is undefined. You can add a guard for it: `promise && promise.then(...)`.

Reply:
Let's change that so `promise` always exists.